pilot/pkg/serviceregistry/kube/controller/ambient/authorization.go

```diff
@@ -265,18 +267,39 @@ func convertPeerAuthentication(rootNamespace string, cfg, nsCfg, rootCfg *securi
 	// Note that this doesn't actually attach the policy to any workload; it just makes it available
 	// to ztunnel in case a workload needs it.
 	foundNonStrictPortmTLS := false
-	for port, mtls := range pa.PortLevelMtls {
+	keys := slices.Sort(maps.Keys(pa.PortLevelMtls))
+	for _, port := range keys {
+		mtls := pa.PortLevelMtls[port]
```

Comment:
This is the backported logic from #54584 for versions of Go before support for iterators, which looks correct to me.